usd/src/usd_parser/USDTransforms_TEST.cc

```diff
@@ -0,0 +1,200 @@
+  }
+
+  EXPECT_EQ(!_rotation.empty(), usdTransforms.Rotation());
+  if (usdTransforms.RotationXYZ() || usdTransforms.RotationZYX())
+  {
+    EXPECT_TRUE(usdTransforms.Rotation());
+  }
+}
+
+/////////////////////////////////////////////////
+TEST(Utils, GetTransform)
+{
+  std::string filename = sdf::testing::TestFile("usd", "upAxisZ.usda");
+  auto stage = pxr::UsdStage::Open(filename);
+  ASSERT_TRUE(stage);
+
+  checkTransforms(
+    "/shapes/ground_plane",
+    stage,
+    ignition::math::Vector3d(0, 0, -0.125),
+    {
+      ignition::math::Quaterniond(1, 0, 0, 0),
+      ignition::math::Quaterniond(1, 0, 0, 0),
+      ignition::math::Quaterniond(1, 0, 0, 0)
+    },
+    ignition::math::Vector3d(1, 1, 1)
+  );
+
+  checkTransforms(
+    "/shapes/ground_plane/link/visual/geometry",
+    stage,
+    ignition::math::Vector3d(0, 0, 0),
+    {},
+    ignition::math::Vector3d(100, 100, 0.25)
+  );
+
+  checkTransforms(
+    "/shapes/cylinder",
+    stage,
+    ignition::math::Vector3d(0, -1.5, 0.5),
+    {
+      ignition::math::Quaterniond(1, 0, 0, 0),
+      ignition::math::Quaterniond(1, 0, 0, 0),
+      ignition::math::Quaterniond(1, 0, 0, 0)
+    },
+    ignition::math::Vector3d(1, 1, 1)
+  );
+
+  checkTransforms(
+    "/shapes/sphere",
+    stage,
+    ignition::math::Vector3d(0, 1.5, 0.5),
+    {
+      ignition::math::Quaterniond(1, 0, 0, 0),
+      ignition::math::Quaterniond(1, 0, 0, 0),
+      ignition::math::Quaterniond(1, 0, 0, 0)
+    },
+    ignition::math::Vector3d(1, 1, 1)
+  );
+
+  checkTransforms(
+    "/shapes/capsule",
+    stage,
+    ignition::math::Vector3d(0, -3.0, 0.5),
+    {
+      ignition::math::Quaterniond(1, 0, 0, 0),
+      ignition::math::Quaterniond(1, 0, 0, 0),
+      ignition::math::Quaterniond(1, 0, 0, 0)
+    },
+    ignition::math::Vector3d(1, 1, 1)
+  );
+
+  checkTransforms(
+    "/shapes/capsule/capsule_link/capsule_visual",
+    stage,
+    ignition::math::Vector3d(0, 0, 0),
+    {
+      ignition::math::Quaterniond(1, 0, 0, 0),
+      ignition::math::Quaterniond(1, 0, 0, 0),
+      ignition::math::Quaterniond(0, 0, M_PI_2)
+    },
+    ignition::math::Vector3d(1, 1, 1)
+  );
+
+  checkTransforms(
+    "/shapes/ellipsoid",
+    stage,
+    ignition::math::Vector3d(0, 3.0, 0.5),
+    {
+      ignition::math::Quaterniond(1, 0, 0, 0),
+      ignition::math::Quaterniond(1, 0, 0, 0),
+      ignition::math::Quaterniond(1, 0, 0, 0)
+    },
+    ignition::math::Vector3d(1, 1, 1)
+  );
+
+  checkTransforms(
+    "/shapes/ellipsoid/ellipsoid_link/ellipsoid_visual/geometry",
+    stage,
+    ignition::math::Vector3d(0, 0, 0),
+    {},
+    ignition::math::Vector3d(0.4, 0.6, 1)
+  );
+
+  checkTransforms(
+    "/shapes/sun",
+    stage,
+    ignition::math::Vector3d(0, 0, 10),
+    {
+      ignition::math::Quaterniond(1, 0, 0, 0),
+      ignition::math::Quaterniond(1, 0, 0, 0),
+      ignition::math::Quaterniond(1, 0, 0, 0)
+    },
+    ignition::math::Vector3d(1, 1, 1)
+  );
```

Comment:
I would like to see a test for objects with non-trivial rotations (with non-zero values for roll pitch and yaw).

Reply:
Thanks for adding the test with non-trivial rotation. However, it still doesn't test how `RotationXYZ` and `RotationZYX` are different. If you converted this `.usda` file to SDFormat, what would be the `<pose>` of the `ellipsoid`? How about if you had `rotateZYX ` instead of `rotateXYZ` in the `.usda` file? Euler angle conventions are always confusing so it's best to iron it out before going forward.